docs/examples.md

```diff
@@ -587,6 +587,218 @@ if ($responseHmac !== $response->getHeader('signature')[0]) {
 echo "\n\nRequest ID: {$response->getHeader('cof-request-id')[0]}\n\n";
 ```
 
+### Calculating HMAC (C#)
+
+```cs
+// HMACCalculation.cs
+using System.Collections.Specialized;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace HMACCalculation
+{
+    public class Crypto
+    {
+        /// <summary>
+        /// Calculate SHA Hash
+        /// </summary>
+        /// <param name="message">Raw string</param>
+        /// <param name="secret">Shared secret</param>
+        /// <param name="encType">encryption Type: sha256 or sha512</param>
+        /// <returns>string</returns>
+        private static string ComputeShaHash(string message, string secret, string encType)
+        {
+            var key = Encoding.UTF8.GetBytes(secret);
+            string outMsg = "";
+            if (encType.Equals("sha512"))
+            {
+                using (var hmac = new HMACSHA512(key))
+                {
+                    var hash = hmac.ComputeHash(Encoding.UTF8.GetBytes(message));
+                    outMsg = BitConverter.ToString(hash).Replace("-", "").ToLower();
+                }
+            }
+            else
```

Comment:
Actually it could be filtered in the beginning of the method and throw an exception if unsupported encType is used. 
in the class:
```C#
static readonly string[] supportedEnc = { "sha256", "sha512" };
```

in the method:
```C#
  if(!Crypto.supportedEnc.Any( e => e.Equals(encType, StringComparison.InvariantCultureIgnoreCase)))
  {
                throw new Exception("Not supported encryption");
  }
```